enhancements/sig-architecture/14-addon-cluster-proxy/README.md

```diff
@@ -0,0 +1,536 @@
+ | -----------------   ---------------     |     ‖
+ | | KubeApiserver |< <| Proxy Agent | <==========
+ | -----------------   ---------------     |
+ -------------------------------------------
+
+```
+
+- A proxy-agent that establishes a long connection towards the proxy-server
+  and register itself w/ the cluster's identity.
+- A proxy-server that receives handshakes from proxy-agent and maintains 
+  tunnel over the long connection.
+- A proxy-client requesting the target kube-apiserver through the tunnel
+  over either websocket HTTP-CONNECT or grpc protocol.
+- An addon controller automates the installation of proxy servers and 
+  proxy agents by consuming the corresponding configuration custom resource.
+  Optionally the proxy-agents can aggregate and report the status upwards
+  so that admins can checkout their healthiness easily w/o even touching
+  the managed-clusters.
+- An addon agent that oversees the proxy-agents and manages the rotation
+  of either certificates or tokens.
+
+
+#### Components
+
+##### Cluster Proxy Operator
+
+According to the latest design and implementation of [addon-framework](https://github.com/open-cluster-management-io/enhancements/tree/main/enhancements/sig-architecture/8-addon-framework),
+we will bring a new agent controller named `cluster-proxy-operator` that
+manages the lifecycle of the proxy components by watching the configuration
+under a new group of api `proxy.open-cluster-management.io/v1alpha1` along
+with a few builtins:
+
+- __ManagedProxyConfiguration__: the configuration template for installing 
+    proxy-servers and proxy-agents.
+
+- __ClusterManagementAddon__: if present, triggering the operator to install
+    the proxy-servers by reading the configuration template in the hub cluster.
+  
+- __ManagedClusterAddon__: if present in any cluster's namespace, prescribe
+    the operator to install proxy-agents according to the template.
+
+Additionally, to make it easier for concrete kubernetes clients to access the
+kube-apiserver in each managed cluster. the operator will be list-watching
+another optional api named `CredentialProjection` to request a dedicated
+service-account token with the requested rbac permission so the actual k8s
+client simply pick up the token then raise request againt `kubernetes.default`
+through the tunnel.
+
+##### ManagedProxyServer
+
+The `ManagedProxyServer` is supposed to be a cluster-scoped resource and is a 
+profile storing configuration template for prescribing the installation of 
+proxy-servers and proxy-agents.
+
+```yaml
+apiVersion: proxy.open-cluster-management.io/v1alpha1
+kind: ManagedProxyConfiguration
+metadata:
+  name: default # Or any unique name, will be the name of the workload
+spec:
+  certificates:
+    signing:
+      additionalSANs:
+        - "1.1.1.1"
+        - "example.com"
+      rotation:
+        expiringDays: 365
+        forceReloadDays: 200
+    bootstrap:
+      secret:
```

Comment:
this might not be needed. registration can help each addon to generate a csr with customized signer. So it is not needed to provide an additional bootstrap secret.

Reply:
oh true, we can reuse the bootstrap secret from registration agent